packages/components/src/components/domainproperties/samples/SampleTypeDesigner.tsx

```diff
@@ -0,0 +1,394 @@
+
+        let parentAliases = Map<string, IParentAlias>();
+
+        if (model && model.importAliases)
+        {
+            let initialAlias = Map<string,string>(model.importAliases);
+            initialAlias.forEach((val, key) => {
+                const newId = SampleTypeDesigner.generateAliasId();
+                parentAliases = parentAliases.set(newId, {
+                    id: newId,
+                    alias: key,
+                    parentValue: options.find(opt => opt.value === val),
+                    ignoreAliasError: false,
+                    ignoreSelectError: false,
+                } as IParentAlias);
+            });
+        }
+
+        this.setState(() => ({
+            parentOptions: options,
+            model: model.merge({parentAliases}) as SampleTypeModel
+        }));
+    };
+
+
+
+    //Generates a temporary id for add/delete of the import aliases
+    static generateAliasId() {
+        return generateId("sampleset-parent-import-alias-");
+    }
+
+    static getImportAliasesAsMap(model: SampleTypeModel): Map<string,string> {
+        const {name, parentAliases } = model;
+
+
+        let aliases = {};
+
+        if (parentAliases) {
+            parentAliases.map((alias: IParentAlias) => {
+                const {parentValue} = alias;
+                let value = parentValue.value as string;
+                if (parentValue === NEW_SAMPLE_SET_OPTION)
+                    value = SAMPLE_SET_IMPORT_PREFIX + name;
+
+                aliases[alias.alias] = value;
+            });
+        }
+
+        return Map<string,string>(aliases);
+    };
+
+    onFieldChange = (newModel: SampleTypeModel) => {
+        this.setState(()=>({model:newModel}));
+    };
+
+    parentAliasChange = (id:string, field: string, newValue: any) => {
+        const {model} = this.state;
+        const {parentAliases} = model;
+        const changedAlias = {
+            ...parentAliases.get(id),
+            [field]: newValue,
+        };
+
+        const newAliases = parentAliases.set(id, changedAlias);
+        const newModel = model.merge({parentAliases: newAliases}) as SampleTypeModel;
+        this.setState(() => ({model: newModel}));
+    };
+
+    addParentAlias = (id:string, newAlias: IParentAlias): void => {
+        const {model} = this.state;
+        let {parentAliases} = model;
+        const newModel = model.merge({parentAliases:parentAliases.set(id, newAlias)}) as SampleTypeModel;
+        this.setState(() => ({model: newModel}));
+    };
+
+    removeParentAlias = (id:string) => {
+        const {model} = this.state;
+        let {parentAliases} = model;
+        const aliases = parentAliases.delete(id);
+        const newModel = model.set('parentAliases', aliases) as SampleTypeModel;
+        this.setState(() => ({model: newModel}));
+    };
+
+    renderDetailsPanel = () => {
+        const {noun = 'Sample Type', nameExpressionInfoUrl, nameExpressionPlaceholder } = this.props;
```

Comment:
Isn't this default value for the noun prop set up in the defaultProps? I'm curious why it is reset / changed here.

Reply:
removed.